test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -4584,6 +4584,34 @@ TEST_P(SslSocketTest, RevokedIntermediateCertificateCRLInTrustedCA) {
   testUtil(complete_unrevoked_test_options.setExpectedSerialNumber(TEST_SAN_DNS4_CERT_SERIAL));
 }
 
+TEST_P(SslSocketTest, RevokedLeafCertificateOnlyLeafCRLValidation) {
```

Comment:
What is this test actually testing? Can you add more comments? Is this expected to succeed when it otherwise fails?